File: How_To_Program_Exercises/JavaHowToProgram9/Chapter3/AccountClass/Account.java
//
//	Account Class
//

public class Account
{
	private double balance;
	
	public Account (double initialBalance)
	{
		//validate initialBalance
		if ( initialBalance > 0.0 )
		{
			balance = initialBalance;
		}
	}
	
	public void credit (double amount)
	{
		balance = balance + amount;
	}
	
	public void debit (double amount)
	{
		if  (amount >= balance)
			balance = balance - amount;
		else {
			System.out.println("Debit amount exceeded account balance.");
		}
			
		
	}
	
	public double getBalance()
	{
		return balance;
	}
}

File: How_To_Program_Exercises/Visual_C#_How_To_Program2012_Fifth/Chapter4/AccountClass/Account.cs
//
//	Account Class Definition
//
using System;
public class Account
{
	private decimal balance;	
	
	//constructor
	public Account (decimal initialBalance)
	{
		Balance = initialBalance;
	}
	
	//Credit an amount to the account
	public void Credit (decimal amount)
	{
		Balance = Balance + amount;
	}
	
	//Method added. 
	public void Debit (decimal amount)
	{
		if ( amount <= Balance)
			Balance = amount;
		else {
			Console.WriteLine("Debit amount exceeded account balance.");
		}
	}
	
	//a property to get and set the account balance
	public decimal Balance
	{
		get { return balance; }
		set {
			//validate value before assigning it to balance
			if (value >= 0)
				balance = value;
		}
	}
}